conda/env/installers/conda.py

```diff
@@ -29,22 +48,90 @@ def _solve(prefix, specs, args, env, *_, **kwargs):
+) -> Environment:
+    """Do a dry run of the environment solve.
+
+    :param specs: Package specifications to install
+    :param args: Command-line arguments
+    :param env: Environment object
+    :return: Solved environment object
+    :rtype: Environment
+    """
     solver = _solve(tempfile.mkdtemp(), specs, args, env, *_, **kwargs)
     pkgs = solver.solve_final_state()
-    solved_env = Environment(
+    return Environment(
         name=env.name, dependencies=[str(p) for p in pkgs], channels=env.channels
     )
-    return solved_env
 
 
-def install(prefix, specs, args, env, *_, **kwargs):
-    """Install packages into an environment"""
+def install(
+    prefix: str, specs: list, args: Namespace, env: Environment, *_, **kwargs
+) -> dict | None:
+    """Install packages into a conda environment.
+
+    This function handles two main paths:
+    1. For environments with explicit_specs (from @EXPLICIT files): Bypasses the solver
+       and directly installs packages using conda.misc.explicit() as required by CEP-23.
+    2. For regular Environment instances: Uses the solver to determine the optimal
+       package set before installation.
+
+    :param prefix: The target installation path for the environment
+    :type prefix: str
+    :param specs: Package specifications to install
+    :type specs: list
+    :param args: Command-line arguments from the conda command
+    :type args: Namespace
+    :param env: Environment object containing dependencies and channels
+    :type env: Environment
+    :return: Installation result information
+    :rtype: dict
+
+    .. note::
+        This implementation follows CEP-23, which states: "When an explicit input file is
+        processed, the conda client SHOULD NOT invoke a solver."
+    """
+    # Handle explicit environments separately per CEP-23 requirements
+    if env.dependencies.explicit:
+        from ...misc import explicit
+
+        log = logging.getLogger(__name__)
+
+        # Use verbose output if not in quiet mode
+        verbose = not context.quiet
+
+        # Determine which package specs to use (priority order):
+        explicit_specs = None
+        filename = env.filename
+
+        # 1. Try to read from original file if available (most reliable source)
+        if filename:
+            explicit_specs = list(yield_lines(filename))
+            if explicit_specs:
+                log.debug(f"Using package specs from explicit file: {filename}")
+            else:
+                log.warning(f"Could not read explicit file {filename} or file is empty")
+
+        # 2-3. Fall back to provided specs or dependencies from environment
+        if not explicit_specs:
+            explicit_specs = specs if specs else env.dependencies.raw
+            log_msg = (
+                "Using specs provided to installer"
+                if specs
+                else "Using dependencies from environment"
+            )
+            log.debug(log_msg)
```

Comment:
I don't fully understand the purpose of this fallback. If the explicit_dependencies are empty, there's nothing to install.

What we are missing here is to use the non-explicit specs as the `history` contents, but I think that's missing in the current `explicit()` function API anyway.